File: atcoder/math-and-algorithm/009.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N, S;
    cin >> N >> S;
    vector<int> A;
    int x;
    for (int i=0; i<N; i++) {
        cin >> x;
        A.push_back(x);
    }

    bool solve = false;
    for (int i=0; i<A.size(); i++){
        for (int j=i+1; j<A.size(); j++){
            if (A[i] + A[j] == S) {
                solve = true;
                break;
            }
        }
        if (solve) break;
    }

    if (solve) {
        printf("YES");
    } else {
        printf("No");
    }
    
    return 0;
}

/*
    009 - Brute Force 2
    https://atcoder.jp/contests/math-and-algorithm/tasks/math_and_algorithm_i
    
    submit
    https://atcoder.jp/contests/math-and-algorithm/submissions/31265327

    AC x 10
    WA x 14
*/

File: 2022/atcoder/math-and-algorithm/NG/009.py
# 1 <= n <= 60
# 1 <= s <= 10000
n, s = map(int, input().split())
# 1 <= a[i] <= 10000
a = list(map(int, input().split()))

solve = False
for i in range(0, len(a)):
    if a[i] == s:
        solve = True
    for j in range(i+1, len(a)):
        if a[i] + a[j] == s:
            solve = True
        for k in range(j+1, len(a)):
            if a[i] + a[j] + a[k] == s:
                solve = True
            for l in range(k+1, len(a)):
                if a[i] + a[j] + a[k] + a[l] == s:
                    solve = True
                for m in range(l+1, len(a)):
                    if a[i] + a[j] + a[k] + a[l] + a[m] == s:
                        solve = True
                    for n in range(m+1, len(a)):
                        if a[i] + a[j] + a[k] + a[l] + a[m] + a[n] == s:
                            solve = True
                        if solve: break
                    if solve: break
                if solve: break
            if solve: break
        if solve: break
    if solve: break
        
if solve:
    print('Yes')
else:
    print('No')